### PR TITLE
Fix: 3.0.0-rc.9 , useAnimatedScrollHandler with FlashList not working

### LINKED ISSUE #3937: useAnimatedScrollHandler not working on Android with RefreshControl version 2.14.1
### Description

useAnimatedScrollHandler not working on Android with RefreshControl

### Steps to reproduce

const AnimatedFlashList = Animated.createAnimatedComponent(FlashList);
 <AnimatedMasonryFlashList
        ref={refList}
        scrollEventThrottle={16}
        data={[...posts]}
        refreshControl={
          <RefreshControl
            onRefresh={() => {
              try {
                onRefresh();
                refetch();
              } catch (x) {}
            }}
            title={t('pull-to-refresh').toString()}
            progressViewOffset={
              Platform.OS === 'ios' ? undefined : HEADER_PROFILE_HEIGHT
            }
            refreshing={isRefreshing}
          />
        }
        numColumns={3}
        contentContainerStyle={flatenContentContainerStyle}
        estimatedItemSize={100}
        renderItem={renderPostItem}
      />

### Snack or a link to a repository

https://gist.github.com/hamdij0maa/b6538a211b35e7f7b3ca1650b727d624

### Reanimated version

2.14.1

### React Native version

0.70+

### Platforms

Android

### JavaScript runtime

Hermes

### Workflow

None

### Architecture

Paper (Old Architecture)

### Build type

Debug mode

### Device

Android emulator

### Device model

Pixel

### Acknowledgements

Yes

### PR DESCRIPTION
**Pull request by orekasep**
## Summary

Fixes #3937.

### CODE REVIEW

**Review by hamdij0maa — APPROVED** [review]

File: src/createAnimatedComponent.tsx [hunk 1]
```diff
@@ -196,7 +196,7 @@ export default function createAnimatedComponent(
 
       if (componentName?.endsWith('FlashList') && this._component) {
         // @ts-ignore it's FlashList specific: https://github.com/Shopify/flash-list/blob/218f314e63806b4fe926741ef73f8b9cd6ebc7eb/src/FlashList.tsx#L824
```
Comment (graszka22) [inline src/createAnimatedComponent.tsx:198]: update the comment

**Review by graszka22 — APPROVED** [review]
Thank you, it fixes the issue 👍 Just update the comment please.

File: src/createAnimatedComponent.tsx (hunk 1 above)
Comment (tomekzaw) [inline src/createAnimatedComponent.tsx:198]:
```suggestion
        // @ts-ignore it's FlashList specific: https://github.com/Shopify/flash-list/blob/218f314e63806b4fe926741ef73f8b9cd6ebc7eb/src/FlashList.tsx#L815
```

Comment (orekasep) [reply src/createAnimatedComponent.tsx:198]: Thanks, I will update shortly

**Review by graszka22 — APPROVED** [review]

**Review by tomekzaw — APPROVED** [review]

**Comment by orekasep** [comment]
Waiting for a merge

**Comment by hamdij0maa** [comment]
still not working on React native 0.71.0

**Comment by tomekzaw** [comment]
> still not working on React native 0.71.0

What exactly is the problem? Which version are you using?

**Review by piaskowyk — APPROVED** [review]

**Comment by hamdij0maa** [comment]
> > still not working on React native 0.71.0
> 
> What exactly is the problem? Which version are you using? I think you confirmed that this PR works for you in [#3937 (comment)](https://github.com/software-mansion/react-native-reanimated/issues/3937#issuecomment-1381548401)

reanimated : 2.14.1 bug on android, RN 0.71.0

**Comment by NTag** [comment]
Actually it works in debug, but on a production build (minified) it doesn't work, the onScroll handler is never called 🥲

**Comment by tomekzaw** [comment]
@NTag that's bad news, can you please check if #3988 fixes the release mode?

**Comment by NTag** [comment]
I only test the rc10, it may have been fixed then! Happy to try once the latest commits are in a rc11.

**Comment by tomekzaw** [comment]
@NTag you can always try nightly build from our main branch: https://github.com/software-mansion/react-native-reanimated/actions/runs/3986157247

**Pull request merged**